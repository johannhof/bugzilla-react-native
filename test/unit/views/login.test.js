import "react-native";
import React from "react";
jest.mock("react-native-config", () => ({API_URL: "bugzilla.api"}));
import Login from "../../../views/login";

// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";

it("renders correctly", () => {
 const tree = renderer.create(<Login />).toJSON();
 expect(tree).toMatchSnapshot();
});

it("renders errors correctly", () => {
 const tree = renderer.create(<Login error={{message: "What the hell"}} />).toJSON();
 expect(tree).toMatchSnapshot();
});
