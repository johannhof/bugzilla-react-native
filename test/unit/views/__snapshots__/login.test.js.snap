exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
      "marginHorizontal": 10
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Enter your BMO username (your email)
  </Text>
  <TextInput
    autoCapitalize="none"
    autoCorrect={false}
    enablesReturnKeyAutomatically={true}
    onChangeText={[Function onChangeText]}
    onSubmitEditing={[Function onSubmitEditing]}
    placeholder="Username"
    returnKeyType="next"
    style={
      Object {
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 40,
        "marginBottom": 10,
        "marginTop": 5,
        "paddingHorizontal": 5
      }
    }
    value="" />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Enter your BMO API Key
  </Text>
  <TextInput
    autoCapitalize="none"
    autoCorrect={false}
    enablesReturnKeyAutomatically={true}
    onChangeText={[Function onChangeText]}
    onSubmitEditing={[Function bound _submit]}
    placeholder="API Key"
    returnKeyType="done"
    style={
      Object {
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 40,
        "marginBottom": 10,
        "marginTop": 5,
        "paddingHorizontal": 5
      }
    }
    value="" />
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent"
        },
        undefined
      ]
    }
    testID="Submit">
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "gray",
          "borderWidth": 1,
          "flex": 1,
          "height": 30,
          "justifyContent": "center",
          "paddingHorizontal": 50
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail">
        Submit
      </Text>
    </View>
  </View>
</View>
`;

exports[`test renders errors correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
      "marginHorizontal": 10
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    testID="errorMessage">
    What the hell
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Enter your BMO username (your email)
  </Text>
  <TextInput
    autoCapitalize="none"
    autoCorrect={false}
    enablesReturnKeyAutomatically={true}
    onChangeText={[Function onChangeText]}
    onSubmitEditing={[Function onSubmitEditing]}
    placeholder="Username"
    returnKeyType="next"
    style={
      Object {
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 40,
        "marginBottom": 10,
        "marginTop": 5,
        "paddingHorizontal": 5
      }
    }
    value="" />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Enter your BMO API Key
  </Text>
  <TextInput
    autoCapitalize="none"
    autoCorrect={false}
    enablesReturnKeyAutomatically={true}
    onChangeText={[Function onChangeText]}
    onSubmitEditing={[Function bound _submit]}
    placeholder="API Key"
    returnKeyType="done"
    style={
      Object {
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 40,
        "marginBottom": 10,
        "marginTop": 5,
        "paddingHorizontal": 5
      }
    }
    value="" />
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent"
        },
        undefined
      ]
    }
    testID="Submit">
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "gray",
          "borderWidth": 1,
          "flex": 1,
          "height": 30,
          "justifyContent": "center",
          "paddingHorizontal": 50
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail">
        Submit
      </Text>
    </View>
  </View>
</View>
`;
